src/components/Word.tsx

```diff
@@ -1,30 +1,60 @@
-import { FC } from "react";
-import { Letter, LetterState } from "./Letter";
+import React from "react";
+import { Letter, LetterState } from "components/Letter";
 
 interface IWord {
-  word: string;
-  typed: string;
+  text: string;
+  value?: string[];
+  isComplete?: boolean;
 }
 
-export const Word: FC<IWord> = ({ word, typed }) => {
-  const letters = word.split("");
+interface IStateWord {
+  element: {
+    index: number;
+    char: string;
+    isHit?: boolean;
+  }[];
+}
```

Comment:
I think the use of an interface is wrong/kinky iffy here.
Maybe replace with just `type Element: { index: number; char: string; isHit?: boolean; }`
And then on line 22 you can just do `let spreadTyped: Element[] = []`

Some thing in WordList for `IStateWordList`

Reply:
See commit 8a0379f for a proposed solution.